Include newsletter

hngerebara opened the pull request:
Please view behavior [here](https://6dd3ffcc.ngrok.io/guides/) 
Try subscribing as well.

https://drive.google.com/file/d/1svUYMclM88880paBVAynbyTRGple6-4W/view?usp=sharing

PS. The URL for displaying the confirmation page will be changed to our website after it's been merged. this is only for testing.

brikis98 commented:
Is the URL after subscribing something we can configure dynamically or only statically in MailChimp? If static, send users to the home page with some thank you dialog; if dynamic, send users back to whatever page they were on when subscribing.

hngerebara commented:
We can configure dynamically in MailChimp so I'll do the latter

hngerebara commented:
This is now ready for review
https://drive.google.com/file/d/1svUYMclM88880paBVAynbyTRGple6-4W/view?usp=sharing

Etiene commented:
Just a small nitpick, the subscribe button can be one more pixel higher

brikis98 commented on _includes/newsletter-modal.html:
```diff
@@ -4,8 +4,8 @@
-                <p>You signed up for our newsletter with the email address: </p>
-                <button type="button" class="btn btn-secondary" data-dismiss="modal">Got it</button>
+                <p>You signed up for our newsletter </p>
+                <button type="button" class="btn btn-secondary" onclick="window.history.pushState({path: window.location.pathname },'',window.location.pathname)" data-dismiss="modal">Got it</button>
```
Why do you need this `onclick` behavior here?

hngerebara replied:
I added it to clear the URL to point to the current path once the button is clicked
E.g When we have the additional parameters in the URL `newsletter_success=1`, after clicking on the `Got it` button which dismisses the modal, the URL remains the same having the url like `/guides?newsletter_success=1. So I did that to ensure that only the pathname is displayed without the parameters.

brikis98 replied:
I see. Can you add that explanation as a comment?

brikis98 commented on _includes/newsletter.html:
```diff
@@ -1,24 +1,15 @@
-                            </button>
-                        </div>
-                    </div>
+            <form  id="guides-newsletter-form"class="col-xs-10 col-xs-offset-1" method="post" action="//gruntwork.us13.list-manage.com/subscribe/post?u=8ccaad0747ac13944cf318e0f&amp;id=c19e11ead2" target="_blank">
```
I noticed the `id=...` param changed. Is that intentional?

hngerebara replied:
Yes, it was. I wanted to specify that this is coming from the guides page. In case we want to make reference to it anytime.

brikis98 commented on _includes/search.html:
```diff
@@ -1,4 +1,4 @@
-<div class="row">
+<div class="row" id="filter-section">
```
Is this change related to this PR?

hngerebara replied:
I included it when I was trying to hide sections before I clarified during the standup. I'll take it out now

brikis98 commented on assets/js/guides.js:
```diff
@@ -1,4 +1,9 @@
+
+  if (window.location.href.indexOf('newsletter_success') !== -1) {
+    $("#newsletter-success").modal('show');
+  } 
```
This check should be in `main.js` so it affects all pages, not just guides.

hngerebara replied:
Okay

brikis98 commented on pages/guides/_no-search-results.html:
```diff
@@ -1,12 +1,8 @@
-                <a href="/contact"><button type="button" class="btn btn-primary">Contact Us</button></a>
-            </div>
-        </div>
+<div id="no-matches" class="no-results no-results-space center">
```
This change doesn't seem related to this PR

hngerebara replied:
No it's not. I noticed it wasn't well-positioned in the middle while I was working on this and decided to fix it at once. I can take it out and create a separate PR.

hngerebara commented:
Ooh crap. I was subscribing them to the `Gruntwork Subscribers Newsletter`. I'll fix that now.

Yes, I fixed the styling issue in this PR. I previously added a `.newsletter` class and styled it mainly for guides page only to realize yesterday that it was the same class name used for styling the /newsletter page. I fixed that already here[](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/177/files#diff-1e69babe6c50a615d3a6ef7b48cf7484R1366). Once this is merged, it will fix that.

hngerebara commented:
Updated this 
https://drive.google.com/file/d/1jj60uhN5fLmCZxl6DXGuwtZ5XqDr83O2/view?usp=sharing
